### PR TITLE
[LTS] Encapsulate sending a communication message 

### PR DESCRIPTION
**Pull request by monassfaro**

### CODE REVIEW

File: xm-commons-mail/src/test/java/com/icthh/xm/commons/mail/provider/communication/CommunicationMailServiceUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    private CommunicationRequestCharacteristic createCharacteristic(String name, String value) {
+        CommunicationRequestCharacteristic characteristic = new CommunicationRequestCharacteristic();
+        characteristic.setName(name);
+        characteristic.setValue(value);
+        return characteristic;
+    }
+
+    @SneakyThrows
+    private String convertToString(Object object) {
+        return new ObjectMapper().writeValueAsString(object);
```
Comment (sergeysenja1992) [inline xm-commons-mail/src/test/java/com/icthh/xm/commons/mail/provider/communication/CommunicationMailServiceUnitTest.java:89]: autowire objectMapper

File: xm-commons-mail/src/test/java/com/icthh/xm/commons/mail/provider/communication/CommunicationMailServiceUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+    private CommunicationMessage createCommunicationMessage(List<CommunicationRequestCharacteristic> characteristics) {
+        CommunicationMessage communicationMessage = new CommunicationMessage();
+        communicationMessage.setSubject("Subject");
+        communicationMessage.setContent("content");
+        communicationMessage.setCharacteristic(characteristics);
+        return communicationMessage;
+    }
+
+    private TemplateModel createTemplateModel() {
```
Comment (sergeysenja1992) [inline xm-commons-mail/src/test/java/com/icthh/xm/commons/mail/provider/communication/CommunicationMailServiceUnitTest.java:73]: ???

File: xm-commons-mail/src/test/java/com/icthh/xm/commons/mail/provider/communication/CommunicationMailServiceUnitTest.java (hunk 1 above)
Comment (monassfaro) [reply xm-commons-mail/src/test/java/com/icthh/xm/commons/mail/provider/communication/CommunicationMailServiceUnitTest.java]: I autowired mapper in the CommunicationMailService but not in the tests. 
Does it required for the tests? In addition, I did not find configured object mapper in any configurations as well as in tests in xm-commons to be able to autowire it there + we use only one method to convert to string.

File: xm-commons-mail/src/test/java/com/icthh/xm/commons/mail/provider/communication/CommunicationMailServiceUnitTest.java (hunk 2 above)
Comment (monassfaro) [reply xm-commons-mail/src/test/java/com/icthh/xm/commons/mail/provider/communication/CommunicationMailServiceUnitTest.java]: Removed

File: xm-commons-messaging/src/main/java/com/icthh/xm/commons/messaging/communication/CommunicationMessage.java [hunk 3]
```diff
@@ -34,5 +38,29 @@ public class CommunicationMessage {
     private List<Receiver> receiver = new ArrayList<>();
     private Sender sender;
     private List<CommunicationRequestCharacteristic> characteristic;
+
+    public CommunicationMessage setTemplateModel(String templateModel) {
+        CommunicationRequestCharacteristic communicationRequestCharacteristic = new CommunicationRequestCharacteristic();
```
Comment (sergeysenja1992) [inline xm-commons-messaging/src/main/java/com/icthh/xm/commons/messaging/communication/CommunicationMessage.java:43]: I see copy/paste, we need method addCharacteristic

File: xm-commons-mail/src/main/java/com/icthh/xm/commons/mail/provider/communication/CommunicationMailService.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.util.Map;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class CommunicationMailService {
```
Comment (sergeysenja1992) [inline xm-commons-mail/src/main/java/com/icthh/xm/commons/mail/provider/communication/CommunicationMailService.java:18]: I think better to remove work Mail and move to communication commons?

Comment (monassfaro) [reply xm-commons-mail/src/main/java/com/icthh/xm/commons/mail/provider/communication/CommunicationMailService.java]: Done

File: xm-commons-messaging/src/main/java/com/icthh/xm/commons/messaging/communication/CommunicationMessage.java (hunk 3 above)
Comment (monassfaro) [reply xm-commons-messaging/src/main/java/com/icthh/xm/commons/messaging/communication/CommunicationMessage.java]: Extracted to the another method

File: xm-commons-messaging/src/main/java/com/icthh/xm/commons/messaging/communication/CommunicationMessage.java [hunk 5]
```diff
@@ -33,6 +37,28 @@ public class CommunicationMessage {
     private String version;
     private List<Receiver> receiver = new ArrayList<>();
     private Sender sender;
-    private List<CommunicationRequestCharacteristic> characteristic;
+    private List<CommunicationRequestCharacteristic> characteristic = new ArrayList<>();
+
+    public CommunicationMessage setTemplateModel(String templateModel) {
```
Comment (sergeysenja1992) [inline xm-commons-messaging/src/main/java/com/icthh/xm/commons/messaging/communication/CommunicationMessage.java:42]: create separate builder without modification of this class

File: settings.gradle [hunk 6]
```diff
@@ -1,3 +1,4 @@
+include ':xm-commons-communication'
```
Comment (sergeysenja1992) [inline settings.gradle:1]: we don't need separate project, put to messaging commons

Comment (monassfaro) [reply settings.gradle]: Done

File: xm-commons-messaging/src/main/java/com/icthh/xm/commons/messaging/communication/CommunicationMessage.java (hunk 5 above)
Comment (monassfaro) [reply xm-commons-messaging/src/main/java/com/icthh/xm/commons/messaging/communication/CommunicationMessage.java]: Created separate class

File: xm-commons-messaging/src/main/java/com/icthh/xm/commons/messaging/communication/CommunicationMessage.java [hunk 7]
```diff
@@ -33,6 +37,7 @@ public class CommunicationMessage {
     private String version;
     private List<Receiver> receiver = new ArrayList<>();
     private Sender sender;
-    private List<CommunicationRequestCharacteristic> characteristic;
+    private List<CommunicationRequestCharacteristic> characteristic = new ArrayList<>();
```
Comment (sergeysenja1992) [inline xm-commons-messaging/src/main/java/com/icthh/xm/commons/messaging/communication/CommunicationMessage.java:40]: rollback

File: xm-commons-messaging/src/main/java/com/icthh/xm/commons/messaging/communication/CommunicationMessage.java [hunk 8]
```diff
@@ -6,6 +6,10 @@
 import java.util.ArrayList;
 import java.util.List;
 
+import static com.icthh.xm.commons.messaging.communication.CommunicationConstants.LANGUAGE;
+import static com.icthh.xm.commons.messaging.communication.CommunicationConstants.TEMPLATE_MODEL;
+import static com.icthh.xm.commons.messaging.communication.CommunicationConstants.TEMPLATE_NAME;
```
Comment (sergeysenja1992) [inline xm-commons-messaging/src/main/java/com/icthh/xm/commons/messaging/communication/CommunicationMessage.java:11]: unused import

File: xm-commons-messaging/src/main/java/com/icthh/xm/commons/messaging/communication/CommunicationCharacteristicsBuilder.java [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+import static com.icthh.xm.commons.messaging.communication.CommunicationConstants.LANGUAGE;
+import static com.icthh.xm.commons.messaging.communication.CommunicationConstants.TEMPLATE_MODEL;
+import static com.icthh.xm.commons.messaging.communication.CommunicationConstants.TEMPLATE_NAME;
+
+@RequiredArgsConstructor
+public class CommunicationCharacteristicsBuilder {
+
+    private final ObjectMapper objectMapper;
+    private final List<CommunicationRequestCharacteristic> characteristics = new ArrayList<>();
```
Comment (sergeysenja1992) [inline xm-commons-messaging/src/main/java/com/icthh/xm/commons/messaging/communication/CommunicationCharacteristicsBuilder.java:19]: CommunicationMessage

addReseiverEmail (String email)

**Review by sergeysenja1992 — APPROVED** [review]

**Review by RomanTeren — APPROVED** [review]

**Pull request merged**